tests/net/socket/tls/src/main.c

```diff
@@ -0,0 +1,244 @@
+    __ASSERT(r != -1, "accept() failed (%d)", r);
+    client_fd = r;
+
+    memset(addrstr, '\0', sizeof(addrstr));
+    addrstrp = (char *)inet_ntop(AF_INET, &sa.sin_addr,
+        addrstr, sizeof(addrstr));
+    __ASSERT(addrstrp != NULL, "inet_ntop() failed (%d)", errno);
+
+    NET_DBG("accepted connection from [%s]:%d as fd %d",
+        log_strdup(addrstr), ntohs(sa.sin_port), client_fd);
+
+    NET_DBG("calling recv()");
+    r = recv(client_fd, addrstr, sizeof(addrstr), 0);
+    __ASSERT(r != -1, "recv() failed (%d)", errno);
+    __ASSERT(r == SECRET_SIZE, "expected: %u actual: %u", SECRET_SIZE, r);
+
+    NET_DBG("calling send()");
+    r = send(client_fd, SECRET, SECRET_SIZE, 0);
+	__ASSERT(r != -1, "send() failed (%d)", errno);
+    __ASSERT(r == SECRET_SIZE, "expected: %u actual: %u", SECRET_SIZE, r);
+
+    NET_DBG("closing client fd");
+    r = close(client_fd);
+    __ASSERT(r != -1, "close() failed on the server fd (%d)", errno);
+}
+
+static void test_tls(void)
+{
+	const int yes = true;
+	static const sec_tag_t sec_tag_list[] = {
+		SERVER_CERTIFICATE_TAG,
+	};
+
+	int r;
+	int server_fd;
+	int client_fd;
+	char *addrstrp;
+	k_tid_t server_thread_id;
+	struct sockaddr_in sa;
+	char addrstr[INET_ADDRSTRLEN];
+
+	k_thread_name_set(k_current_get(), "ztest_thread");
+
+	/* Load credentials */
+	NET_DBG("Loading credentials");
+	r = tls_credential_add(SERVER_CERTIFICATE_TAG,
+		TLS_CREDENTIAL_CA_CERTIFICATE,
+		ca, sizeof(ca));
+	zassert_equal(r, 0, "failed to add CA Certificate (%d)", r);
+
+	r = tls_credential_add(SERVER_CERTIFICATE_TAG,
+		TLS_CREDENTIAL_SERVER_CERTIFICATE,
+		server, sizeof(server));
+	zassert_equal(r, 0, "failed to add Server Certificate (%d)", r);
+
+	r = tls_credential_add(SERVER_CERTIFICATE_TAG,
+		TLS_CREDENTIAL_PRIVATE_KEY,
+		server_privkey, sizeof(server_privkey));
+	zassert_equal(r, 0, "failed to add Server Private Key (%d)", r);
+
+	/* Create server socket */
+	NET_DBG("Creating server socket");
+	r = socket(AF_INET, SOCK_STREAM, IPPROTO_TLS_1_2);
+	zassert_not_equal(r, -1, "failed to create server socket (%d)", errno);
+	server_fd = r;
+
+    r = setsockopt(server_fd, SOL_SOCKET, SO_REUSEADDR, &yes, sizeof(yes));
+    zassert_not_equal(r, -1, "failed to set SO_REUSEADDR (%d)", errno);
+
+	r = setsockopt(server_fd, SOL_TLS, TLS_SEC_TAG_LIST,
+		sec_tag_list, sizeof(sec_tag_list));
+	zassert_not_equal(r, -1, "failed to set TLS_SEC_TAG_LIST (%d)", errno);
+
+	memset(&sa, 0, sizeof(sa));
+	r = inet_pton(AF_INET, CONFIG_NET_CONFIG_MY_IPV4_ADDR,
+		&sa.sin_addr.s_addr);
+	sa.sin_family = AF_INET;
+	sa.sin_port = htons(PORT);
+
+	r = bind(server_fd, (struct sockaddr *)&sa, sizeof(sa));
+	zassert_not_equal(r, -1, "failed to bind (%d)", errno);
+
+	r = listen(server_fd, 1);
+	zassert_not_equal(r, -1, "failed to listen (%d)", errno);
+
+    memset(addrstr, '\0', sizeof(addrstr));
+    addrstrp = (char *)inet_ntop(AF_INET, &sa.sin_addr,
+        addrstr, sizeof(addrstr));
```

Comment:
indentation wrong, and actually looks like indentation is incorrect most of the multiline statements, can you recheck all of them